### DISCUSSION TITLE
[Suggestions] Creature recruit window, add a real "CHANGE" button (Discussion)

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2755] [Ideas]
> In fheroes2, it is possible to choose between the base creature and the upgraded creature in the creature recruit window.
> This is a good improvement to the original game.
> However the way it's done with the text over the creature sprite is not optimal.
>
> In fheroes2:
>
> ![Recruit creature fh2 coupé 01](https://user-images.githubusercontent.com/43583994/93664223-7d5f5380-fa6d-11ea-9dda-1492c1d574ea.png)

Latest proposal and feedback from this thread https://github.com/ihhub/fheroes2/issues/1744:

> Hello @Branikolog,
> 
> I think this one could be the right one. :)
> 
> ![Recruit Dragon change button left side modifié 01](https://user-images.githubusercontent.com/43583994/107126216-bf6b2380-68ae-11eb-98e6-5cd28f9e825a.png)

> Hi, @LeHerosInconnu !
> I think we need to make a separate discussion for this issue. ) I think we'd spend lots of time choosing the best solution. :)
> On subject:
> Well, this location for "Change" button is better from the rest, but in this case "Available 0" is definitely not on a proper place.
> Also, I wish to have not such "large" solution of this problem. ) something less noticeable. I mean, this button changes the perception of this window, because it's rather huge.

The position of "Available: 0" doesn't bother me with the cost of the creature just below it.

With a "NEXT" button, or maybe something more like this:

![Recruit Dragon change button left side rectangles upgrade 02](https://user-images.githubusercontent.com/43583994/107133481-df1f3d80-68e8-11eb-9886-212ffca93854.png)

**Edit 2021/02/07.**

Or this one:

![Recruit Dragon change button left side rectangles upgrade 06](https://user-images.githubusercontent.com/43583994/107158526-09293c00-698b-11eb-92f5-d67d524ac6b3.png)

**Edit 2021/02/08.**

Or this other:

![Recruit Dragon change button left side flèches upgrade 08](https://user-images.githubusercontent.com/43583994/107288130-3d702b80-6a63-11eb-8b74-d50369a6d181.png)

**Comment by Branikolog** [comment]
@LeHerosInconnu 
I start feeling really uncomfortably: You're proposing a bunch of ideas and the only thing I do is just criticizing. )
Your last proposal looks much, much better due to the symmetry we watch, regarding Max and Change buttons (As they have the same size.)
Placing "Available" in the upper part makes me again feel compressed.) Placing it in a window "Cost per troop" makes sense, but in this case we have to draw new frames, which could be not easy task currently, I believe. (As well as filling of the new button)
Anyway, I'm on the way to feel compression here on screenshot... :D
![Compression](https://user-images.githubusercontent.com/55348946/107877172-d5469d00-6edb-11eb-9926-d1aeff907b0d.png)

Could you, please, apply some criticism now for my variant, so I don't feel so grumpy person? =)

![Change Arrows](https://user-images.githubusercontent.com/55348946/107924177-99671280-6f83-11eb-8390-0e09d8482e50.png)

I still believe, this implementation is not bad at all:
We apply minimum changes to the window.
Adding these two arrows is quite a simpler task.
I agree, that we can feel a bit of "compression" here... :D But these arrows are rather tiny and are made in yellow colors which makes them much less noticeable.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

I will answer later. :)

**Reply by Branikolog** [reply]
I'm starting feeling nervous now. )

**Comment by ihhub** [comment]
Hi @LeHerosInconnu and @Branikolog , I think we can go with arrows proposed by @Branikolog . We can still keep this topic open as the proposed solution is not ideal yet but I think it is better than the current implementation.

What do you think?

**Reply by LeHerosInconnu** [reply]
Hello @ihhub,

> Hi @LeHerosInconnu and @Branikolog , I think we can go with arrows proposed by @Branikolog . We can still keep this topic open as the proposed solution is not ideal yet but I think it is better than the current implementation.
> 
> What do you think?

I agree with this.

**Comment by ihhub** [comment]
Hi @LeHerosInconnu and @Branikolog , the change was integrated into master branch. We could slightly polish the existing design and come up with a better solution later :)
